compiler/ballerina-lang/src/main/java/io/ballerina/compiler/api/impl/SymbolFactory.java

```diff
@@ -375,8 +389,18 @@ public BallerinaMemberTypeSymbol createTupleMember(BVarSymbol symbol) {
         return new BallerinaMemberTypeSymbol(context, symbol, type);
     }
 
+    /**
+     * Create a named path segment symbol.
+     *
+     * @param symbol {@link BResourcePathSegmentSymbol} to convert
+     * @return {@link BallerinaNamedPathSegment} generated
+     */
+    public BallerinaNamedPathSegment createNamedPathSegment(BResourcePathSegmentSymbol symbol) {
```

Comment:
NamedPathSegment is not something that comes from the spec, right? Isn't there a name we can use that aligns better with the spec?

Reply:
We have already exposed this as "NamedPathSegment", and ballerina extensions have used it as it is.

Reply:
If we change this, it could lead to a break

Reply:
Ack.